components/cdc/src/endpoint.rs

```diff
@@ -594,25 +614,25 @@ impl<T: 'static + RaftStoreRouter<RocksEngine>> Endpoint<T> {
             .iter()
             .map(|(region_id, delegate)| (*region_id, delegate.id))
             .collect();
+        let region_count = regions.len();
+        let tso_worker = self.tso_worker.clone();
+        let scheduler_clone = self.scheduler.clone();
         let fut = tso.join(timeout.map_err(|_| unreachable!())).then(
             move |tso: pd_client::Result<(TimeStamp, ())>| {
                 // Ignore get tso errors since we will retry every `min_ts_interval`.
                 let (min_ts, _) = tso.unwrap_or((TimeStamp::default(), ()));
+                let (tx, rx) = mpsc::channel(region_count);
```

Comment:
Please add an upper limit, otherwise, it may allocate a large chunk of memory and this is not a recommended approach.